main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1389,16 +1389,24 @@ object Weeder2 {
                 case Pattern.Wild(loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
                 case Pattern.Var(ident, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Var(ident, loc))
                 case Pattern.Cst(Ast.Constant.Unit, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
-                case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+                case other =>
+                  val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+                  val ident = Name.Ident(s"x${flix.genSym.freshId()}", other.loc.asSynthetic)
+                  Validation.toSoftFailure(WeededAst.RestrictableChoosePattern.Var(ident, other.loc.asSynthetic), e)
```

Comment:
This should use the error construct right?

Reply:
See the other PR i opened